test/ziggurat/messaging/producer_test.clj

```diff
@@ -56,6 +56,26 @@
         (let [message-from-mq (rmq/get-msg-from-channel-dead-queue topic-entity channel)]
           (is (= expected-message-payload message-from-mq))))))
 
+  (testing "message in channel will be retried in delay queue with suffix 1 if message retry-count exceeds retry count in channel config"
```

Comment:
we should do a similar test for normal retry as well. Basically, add produce and consumer for normal retry in this test